Survey tryout

MaaJoo13 opened the pull request:
WIP

I don't know how commit [259882b] got into this PR, it's not mine...

JulianBarinton committed:
Update README.md

JulianBarinton committed:
Update solver installation link

Bachibouzouk committed:
Process the survey answers

- Changed the subquestion mapping to add the component related to the
question
- Fix the question parsing
- Map the answers to 3 possible cases, component, component attribute
and no match

Co-authored-by: MaaJoo13 <148241644+MaaJoo13@users.noreply.github.com>

Bachibouzouk committed:
Add way to save survey results

Bachibouzouk committed:
Move the csv files to a package form

Previously it was in component_library/data, unsure why WIP_components
was used instead

Bachibouzouk committed:
Fix the component library listing and datapackage

Bachibouzouk committed:
Fix component library structure

The "data" folder was missing

Bachibouzouk committed:
Remove unused code from survey

Bachibouzouk committed:
Fix the mapping of components to resources/category in the library

Each component is a row within a csv file, called a resource from the
datapackage perspective. For adding components automatically it is
important to know in which resource a component is located, thus this
mapping provided by the function `utils.list_available_components()`.
The module can be run standalone `python utils.py` to detect potential
problems each time one introduces a new component into the library.

Bachibouzouk committed:
Add answer component mapping and subquestion mapping generation

To generate the subquestion mapping, one needs to run the
`survey/survey.py` code once

Bachibouzouk committed:
Fix component parsing from the survey's answers

Bachibouzouk committed:
Use datapackage to add components to a new datapackage

Bachibouzouk committed:
Future changes for sequences and busses

Bachibouzouk committed:
Example of survey specific question treatment

For cases were we need to do something depending on another answer
within the survey, it might be easier to hardcode the fonction, this is
just an example which would need to be filled

Bachibouzouk committed:
Remove unused code

´

Bachibouzouk committed:
Add script to generate the answer mapping and subquestion mapping

Bachibouzouk commented:
@JulianBarinton - Now the WIP_components folder is structured as a datapackage,  I put your comments which prevented the smooth readin of the package resources (resource=csv file) in the `comments.md` file and tried to document as much as possible what I did in `component_library/README.md`. Ideally you could support in filling the components into WIP_components reference package and test that for certain survey answer configuration, that the correct components are copied to the scenario datapackage.

Bachibouzouk committed:
How to generate the datapackage.json for component library

Bachibouzouk committed:
First draft of component library datapackage

Bachibouzouk committed:
Update the readme

Bachibouzouk committed:
Copy the busses from the component library thanks to foreignKeys

Bachibouzouk committed:
Add sequence processing

It is not done yet, one need to provide a file with the sequences

Bachibouzouk committed:
Distinguish between inferring foreignKeys and building datapackage.json

Bachibouzouk committed:
Fix format errors in the component library

Also add a mimo in crops

Bachibouzouk committed:
Add a script to run the datapackage and generate a picture with visio

Bachibouzouk committed:
Delete unused and confusing file

The one used is within app folder

Bachibouzouk committed:
Fix commands because sub_question mapping was updated

Bachibouzouk committed:
Fix code which did not work on windows because of \

Bachibouzouk committed:
Add option for "other" in the survey

Bachibouzouk committed:
Add more components to the answers of question 1

Here the problem currently is that the capacity is being assigned to all
the components

Bachibouzouk committed:
Fix component library

Wrong bus names, wrong format of certain columns, etc

Bachibouzouk committed:
Fix warning with concat of dataframes

Bachibouzouk committed:
Log information to busses and profiles added to the energy system

JulianBarinton committed:
answer component mapping (WIP)

JulianBarinton committed:
WIP survey answer component mapping

Bachibouzouk committed:
Add apv to component library ressources

Bachibouzouk committed:
Fix setting components' attributes

Bachibouzouk committed:
Eliminate space for hypdro-power component

Bachibouzouk committed:
Get the weather data from our API

Note: one need to add WEATHER_DATA_API_HOST to the environment variables

Bachibouzouk committed:
Bind the weather data in the scenario building

MaaJoo13 committed:
WIP scenario profile mapping

MaaJoo13 committed:
WIP scenario profile mapping 2

MaaJoo13 committed:
finished scenario profile mapping

Bachibouzouk committed:
Change the keys of the components attribute of the ScenarioBuilder

Before it was the name of the component from the component library as
key, now it is the name of the component from the component libray and
the name one want to give this component as a user in a tuple:

("septic_tank", "myseptictank1")

("septic_tank", "myseptictank2")

Bachibouzouk committed:
Remove unused code

Bachibouzouk committed:
Fix datapackage.json of WIP component list

Bachibouzouk committed:
Merge pull request #11 from rl-institut/feature/get-weather-profiles

Get the weather data from our API

Bachibouzouk committed:
Merge pull request #12 from rl-institut/feature/enable-multiple-component-of-same-facade

Change the keys of the components attribute of the ScenarioBuilder

MaaJoo13 commented:
Already integrated into https://github.com/rl-institut/WEFEConfigurator/pull/13
